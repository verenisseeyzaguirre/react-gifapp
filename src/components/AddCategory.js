import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState(''); // si no habia '' era undefined y salia error

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value )
        // console.log(e.target.value)
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit llamado', inputValue)

        // cats es el estado anterior
        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [inputValue, ...cats] );
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit= {handleSubmit}>
            <input 
                type="text"
                value = { inputValue }
                onChange = { handleInputChange }
                placeholder= 'Buscar gif...'
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };