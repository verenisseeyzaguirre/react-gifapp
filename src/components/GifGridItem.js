import React from 'react'


import PropTypes from 'prop-types';
//https://animate.style/

// param ( img ) en p {img.title}
export const GifGridItem = ( { title, url } ) => {

    //console.log( img );

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={ url } alt={ title }/>
            <div className='name'>
                <p>{ title }</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

