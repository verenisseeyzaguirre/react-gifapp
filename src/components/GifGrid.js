import React from 'react'
//, { useState, useEffect }
import { GifGridItem } from './GifGridItem';
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( { category } ) => {

    //const [count, setCount] = useState(0);
    //const [images, setImages] = useState([]);

    const { data:images, loading } = useFetchGifs( category );

    // [] indica q se ejecuando cuando se renderiza por primera vez y ya no mas
    // salio un warning porque si cambia la categoria deberia renderizar nuevmnte
    /*
    useEffect( () => {
        getGifs( category )
            .then( setImages );
    }, [ category ])
    */
    // getGifs()

    return (
        <>
            <h3 className='animate__animated animate__fadeIn' > {category} </h3>

            { loading && <p className='animate__animated animate__flash' >Loading</p> }

            <div className='card-grid' >
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    ))
                }         
            </div>
        </>
    )
}

/*
    debajo de h3
    <h2>{count}</h2>
    <button onClick= {()=> setCount(count+1) }>+1</button>
*/

/*
<ol>
    {
        images.map( img => (
            <li  key={img.id}>{img.title}</li>
        ))
    }
</ol>
*/

/*
{
    images.map( img => (
        <GifGridItem 
            key={img.id}
            {...img}
        />
    ))
}  
*/