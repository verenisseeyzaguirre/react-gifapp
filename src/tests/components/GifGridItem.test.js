import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

//import { render } from '@testing-library/react';
/*
const img = {
        id: '123456',
        title: 'titulo',
        url: 'https://localhost/algo.jpg'
    }
*/

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'titul';
    const url = 'https://localhost/algo.jpg'
    

    const wrapper = shallow( <GifGridItem title = {title} url = {url} /> )

    test('Debe mostrar <GifGridItem /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });


    test('Debe tener un parrafo con el title', () => {
        
        const etiquetaP = wrapper.find('p');
        expect( etiquetaP.text().trim() ).toBe(title);

    });
    
    test('Debe tener una imagen con el enlace fuente y nombre alternativo', () => {
        
        const etiquetaImg = wrapper.find('img');
        expect( etiquetaImg.props().src ).toBe(url);
        expect( etiquetaImg.prop('alt') ).toBe(title);

    });

    test('Debe tener animate__fadeIn', () => {
        
        const etiquetaDiv = wrapper.find('div').at(0);
        //const className = etiquetaDiv.prop('className');
        //expect( className.hasClass('animate__fadeIn') ).toBeTruthy();
        expect( etiquetaDiv.hasClass('animate__fadeIn') ).toBeTruthy();

    });

})

