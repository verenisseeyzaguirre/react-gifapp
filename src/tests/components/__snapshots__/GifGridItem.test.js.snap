// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titul"
    src="https://localhost/algo.jpg"
  />
  <div
    className="name"
  >
    <p>
      titul
    </p>
  </div>
</div>
`;
